src/software/backend/replay_logging/replay_reader.cpp

```diff
@@ -0,0 +1,139 @@
+#include "replay_reader.h"
+
+#include <google/protobuf/util/delimited_message_util.h>
+
+#include <fstream>
+
+// unfortunately protobuf forces me to write caveman code
+extern "C"
+{
+#include <fcntl.h>
+#include <unistd.h>
+}
+
+namespace fs = std::experimental::filesystem;
+
+namespace
+{
+    ReplayMsg readDelimitedReplayProtobufFile(const fs::path& file_path)
+    {
```

Comment:
Making `readDelimitedReplayProtobufFile` a private const function or a static function within `ReplayReader`, is more consistent with our coding style. Is there a reason to prefer a function in an anonymous namespace?

Reply:
Made this a static member function of `ReplayReader`